src/fromager/constraints.py

```diff
@@ -66,4 +66,6 @@ def write_from_build_order(filename: pathlib.Path, build_stack: list[typing.Any]
                 f"-{dep_type}-> {req.name}({version})"
                 for dep_type, req, version in step["why"]
             )
+            constraint = f' Constraints: {step["constraint"]}'
+            comment.join(constraint)
```

Comment:
I don't think this is doing what you expect

```
>>> "a".join("b")
'b'
>>>
```

Reply:
Hmm, I have removed the `join()` call and added `constraints` to existing comment.